src/material/button/_button-theme-private.scss

```diff
@@ -18,8 +18,11 @@
     opacity: map.get($opacities, hover);
   }
 
-  &:focus .mat-mdc-button-persistent-ripple::before {
-    opacity: map.get($opacities, focus);
+  .cdk-program-focused,
```

Comment:
Ah actually these selectors should be prefixed with `&`, otherwise the button focus indication will show when it's inside of any focused element.

Reply:
Good catch - done.